asv_bench/benchmarks/Argofloat_benchmark.py

```
@@ -0,0 +1,97 @@
+        if particle.drift_age >= drifttime:
+            particle.drift_age = 0  # reset drift_age for next cycle
+            particle.cycle_phase = 2
+
+    elif particle.cycle_phase == 2:
+        # Phase 2: Sinking further to maxdepth
+        particle_ddepth += vertical_speed * particle.dt
+        if particle.depth + particle_ddepth >= maxdepth:
+            particle_ddepth = maxdepth - particle.depth
+            particle.cycle_phase = 3
+
+    elif particle.cycle_phase == 3:
+        # Phase 3: Rising with vertical_speed until at surface
+        particle_ddepth -= vertical_speed * particle.dt
+        # particle.temp = fieldset.temp[time, particle.depth, particle.lat, particle.lon]  # if fieldset has temperature
+        if particle.depth + particle_ddepth <= fieldset.mindepth:
+            particle_ddepth = fieldset.mindepth - particle.depth
+            # particle.temp = 0./0.  # reset temperature to NaN at end of sampling cycle
+            particle.cycle_phase = 4
+
+    elif particle.cycle_phase == 4:
+        # Phase 4: Transmitting at surface until cycletime is reached
+        if particle.cycle_age > cycletime:
+            particle.cycle_phase = 0
+            particle.cycle_age = 0
+
+    if particle.state == StatusCode.Evaluate:
+        particle.cycle_age += particle.dt  # update cycle_age
+
+
+class ArgoFloatJIT:
+    def setup(self):
+        xdim = ydim = zdim = 2
+
+        dimensions = {
+            "lon": "lon",
+            "lat": "lat",
+            "depth": "depth",
+        }
+        data = {
+            "U": np.ones((xdim, ydim, zdim), dtype=np.float32),
+            "V": np.zeros((xdim, ydim, zdim), dtype=np.float32),
+        }
+        data["U"][:, :, 0] = 0.0
+        fieldset = FieldSet.from_data(data, dimensions, mesh="flat", transpose=True)
+        fieldset.mindepth = fieldset.U.depth[0]
+
+        # Define a new Particle type including extra Variables
+        self.ArgoParticle = JITParticle.add_variables(
```

`ArgoParticle` does not need to be a class attribute. So drop the `self.` here.

---

I think this is what let the benchmark fail? Later the `ArgoParticle` is not defined when the filedset is created.

---

ok!